app/src/androidTest/java/org/mozilla/fenix/ui/robots/SettingsSubMenuSitePermissionsAutoPlayRobot.kt

```diff
@@ -0,0 +1,75 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+@file:Suppress("TooManyFunctions")
+
+package org.mozilla.fenix.ui.robots
+
+/**
+ * Implementation of Robot Pattern for the settings Site Permissions Autoplay sub menu.
+ */
+
+import androidx.test.espresso.Espresso.onView
+import androidx.test.espresso.assertion.ViewAssertions.matches
+import androidx.test.espresso.matcher.ViewMatchers.Visibility
+import androidx.test.espresso.matcher.ViewMatchers.withContentDescription
+import androidx.test.espresso.matcher.ViewMatchers.withEffectiveVisibility
+import androidx.test.espresso.matcher.ViewMatchers.withId
+import androidx.test.platform.app.InstrumentationRegistry
+import androidx.test.uiautomator.UiDevice
+import org.hamcrest.CoreMatchers.allOf
+import org.mozilla.fenix.R
+import org.mozilla.fenix.helpers.assertIsChecked
+import org.mozilla.fenix.helpers.click
+
+class SettingsSubMenuSitePermissionsAutoPlayRobot {
+
+    fun verifyNavigationToolBarHeader() = assertNavigationToolBarHeader()
+
+    fun verifySitePermissionsAutoPlaySubMenuItems() {
+        assertBlockAudioAndVideoOnMobileDataOnlyAudioAndVideoWillPlayOnWiFi()
+        assertBlockAudioOnly()
+        assertVideoAndAudioBlockedRecommended()
+        assertCheckRadioButtonDefault()
+    }
+
+    class Transition {
+        val mDevice = UiDevice.getInstance(InstrumentationRegistry.getInstrumentation())!!
+
+        fun goBack(interact: SettingsSubMenuSitePermissionsRobot.() -> Unit): SettingsSubMenuSitePermissionsRobot.Transition {
+            mDevice.waitForIdle()

```

Comment:
Do we need this line? It is working for me without it...

Reply:
As you know, go back is for returning back to the screen again from sub-activity to main activity again. Here, I have to go back to SitePermissions from AutoPlay. Hence, used it. I wonder why is it working without that.

Reply:
If I'm not mistaken we have several goBack calls in the code already and no need for that waitForIdle...

Reply:
Oh yeah, I was mistaken. I will remove that